deep_qa/models/sequence_tagging/verb_semantics_model.py

```diff
@@ -0,0 +1,57 @@
+from keras.layers import Dense, Input, TimeDistributed, Multiply
+from overrides import overrides
+
+from deep_qa.data.instances.sequence_tagging.verb_semantics_instance import VerbSemanticsInstance
+from deep_qa.common.params import Params
+from deep_qa.data.instances.sequence_tagging import concrete_instances
+from deep_qa.training.text_trainer import TextTrainer
+from deep_qa.training.models import DeepQaModel
+
+class VerbSemanticsModel(TextTrainer):
+    """
+    This ``VerbSemanticsModel`` encodes a sequence of text with some number of stacked
+    ``seq2seq_encoders``, then predicts a tag at each index.
+
+    Parameters
+    ----------
+    num_stacked_rnns : int, optional (default: ``1``)
+        The number of ``seq2seq_encoders`` that we should stack on top of each other before
+        predicting tags.
+    instance_type : str
+        Specifies the particular subclass of ``TaggedSequenceInstance`` to use for loading data,
+        which in turn defines things like how the input data is formatted and tokenized.
+    """
+    def __init__(self, params: Params):
+        self.num_stacked_rnns = params.pop('num_stacked_rnns', 1)
+        instance_type = params.pop('instance_type', "VerbSemanticsInstance")
+        self.instance_type = concrete_instances[instance_type]
+        super(VerbSemanticsModel, self).__init__(params)
+
+    @overrides
+    def _instance_type(self):  # pylint: disable=no-self-use
+        return self.instance_type
+
+    @overrides
+    def _build_model(self):
+        # shape: (batch_size, text_length)
+        text_input = Input(shape=self._get_sentence_shape(), dtype='int32', name='word_array_input')
+
+        verb_input = Input(shape=self._get_sentence_shape(), dtype='int32', name='verb_index_input')
+        entity_input = Input(shape=self._get_sentence_shape(), dtype='int32', name='entity_index_input')
+
+        # shape: (batch_size, text_length, embedding_dim)
+        text_embedding = self._embed_input(text_input)
+        for i in range(self.num_stacked_rnns):
+            encoder = self._get_seq2seq_encoder(name="encoder_{}".format(i),
+                                                fallback_behavior="use default params")
+            # shape still (batch_size, text_length, embedding_dim)
+            text_embedding = encoder(text_embedding)
+        # The -2 below is because we are ignoring the padding and unknown tokens that the
+        # DataIndexer has by default.
+        predicted_tags = TimeDistributed(Dense(self.data_indexer.get_vocab_size('tags') - 2,
+                                               activation='softmax'))(text_embedding)
+        return DeepQaModel(input=text_input, output=predicted_tags)
```

Comment:
This needs to be something like `return DeepQaModel(inputs=[text_input, verb_input, entity_input], outputs=[predicted_tags, state_change])`.